src/components/PageSettingsModal.jsx

```diff
@@ -22,12 +28,38 @@ export default class PageSettingsModal extends Component {
         title: '',
         permalink: '',
       },
+      permalinkSetterIsActive: false,
+      baseUrl: '',
+      pagePermalinks: [],
     };
   }
 
   async componentDidMount() {
     try {
-      const { siteName, fileName, isNewPage } = this.props;
+      const {
+        siteName, fileName, isNewPage, pageFilenames,
+      } = this.props;
+
+      // get settings data from backend
+      const settingsResp = await axios.get(`${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}/settings`, {
+        withCredentials: true,
+      });
+      const { settings } = settingsResp.data;
+      const { configFieldsRequired } = settings;
+      const baseUrl = configFieldsRequired.url;
+      this.setState({ baseUrl });
```

Comment:
I think React batches the setState methods and determines when to render, so each setState call doesn’t result in a re-render, but I’m down to remove that line if it looks verbose. I added it because I thought it would make more sense to set the state next to where baseUrl was defined

Reply:
Just leaving this here for reference: `setState` batches up the actual setting of state, but every time `setState` is called it will be followed by a rerender ([source](https://reactjs.org/docs/react-component.html#setstate))